libmikmod/loaders/load_amf.c

```diff
@@ -114,7 +114,48 @@ static void AMF_Cleanup(void)
 	track=NULL;
 }
 
-static BOOL AMF_UnpackTrack(MREADER* r)
+/* Some older version 1.0 AMFs contain an anomaly where the sample length is
+ * a DWORD, the loop start is a WORD, and the loop end is missing. Later AMF
+ * 1.0 modules and up contain all three as DWORDs, and earlier versions have
+ * all three as WORDs. This function tries to detect this edge case in the
+ * instruments table. This should only be called on 1.0 AMFs.
+ */
+static BOOL AMF_ScanV10Instruments(MREADER *r, unsigned int numins)
+{
+	LONG resetpos;
```

Comment:
This should be `SLONG`

Reply:
Whoops, that should be fixed now. Since MikMod uses Windows.h types everywhere I assumed that was one of them, but I was wrong!